src/kem/kyber.rs

```diff
@@ -133,7 +136,11 @@ pub(super) fn encapsulate<
         ETA2_RANDOMNESS_SIZE,
     >(public_key.as_slice(), randomness, pseudorandomness);
 
-    (ciphertext.into(), shared_secret.try_into().unwrap())
+    let shared_secret = match shared_secret.try_into() {
+        Ok(shared_secret) => shared_secret,
+        Err(_) => panic!(),
```

Comment:
should this panic or return an Error?

Reply:
This should indeed panic because it can not fail (as proven in F* as well, Rust just doesn't know it unfortunately). The panic is just explicit here (instead of the unwrap) because of some shortcomings in the tooling we work around for now.